scripts/ansible-todd-dev.yml

```diff
@@ -1,83 +1,135 @@
+      stat:
+        path: "{{ todd_make_configureenv_check_file }}"
+      register: "make_configureenv_check"
+
+    - name: Configure Environment
+      command: "make configureenv" #what is changed in order to determine if this needs to run again
+      args:
+        chdir: "{{ todd_dir }}"
+      register: "make_configureenv"
+      when: not make_configureenv_check.stat.exists
+
+    - name: Marking Configure Environment As Complete
+      file:
+        path: "{{ todd_make_configureenv_check_file }}"
+        state: "touch"
+      when: make_configureenv.changed
+
+    - name: GOPATH Owner  #Needs to be changed to ensure idempotency
+#      file:
+#        path: "/home/vagrant/go"
+#        owner: "vagrant"
+#        group: "vagrant"
+#        recurse: yes
+#        state: "directory"
+      shell: chown vagrant:vagrant $(find /home/vagrant/go | grep -v todd) warn=no
 
 - name: Prepping for ToDD
   hosts: all
   vars:
-    todd_dir: /home/vagrant/go/src/github.com/Mierdin/todd
+    todd_compile_check_file: '/var/log/.todd_compile_complete'
+    todd_dir: '/home/vagrant/go/src/github.com/Mierdin/todd'
   tasks:
 
-  - name: Set GOPATH for Vagrant user profile
-    lineinfile: 
-      dest=/home/vagrant/.profile
-      state=present
-      regexp='^export GOPATH'
-      line='export GOPATH=/home/vagrant/go'
-
-  - name: Set PATH for Vagrant user profile
-    lineinfile: 
-      dest=/home/vagrant/.profile
-      state=present
-      regexp='^export PATH'
-      line='export PATH=$PATH:/home/vagrant/go/bin:/usr/local/go/bin:{{ todd_dir }}/scripts'
-
-  - name: cd to todd_dir when logging in
-    lineinfile: 
-      dest=/home/vagrant/.profile
-      state=present
-      regexp='^cd '
-      line='cd {{ todd_dir }}'
-
-  - name: Compile ToDD
-    shell: . /home/vagrant/.profile && make chdir={{ todd_dir }}
+    - name: Set GOPATH for Vagrant user profile
+      lineinfile:
+        dest: "/home/vagrant/.profile"
+        state: "present"
+        regexp: "^export GOPATH"
+        line: "export GOPATH=/home/vagrant/go"
+
+    - name: Set PATH for Vagrant user profile
+      lineinfile:
+        dest: "/home/vagrant/.profile"
+        state: "present"
+        regexp: "^export PATH"
+        line: "export PATH=$PATH:/home/vagrant/go/bin:/usr/local/go/bin:{{ todd_dir }}/scripts"
+
+    - name: cd to todd_dir when logging in
+      lineinfile:
+        dest: "/home/vagrant/.profile"
+        state: "present"
+        regexp: "^cd "
+        line: "cd {{ todd_dir }}"
+
+    - name: Checking Status of Compile ToDD
+      stat:
+        path: "{{ todd_compile_check_file }}"
+      register: "todd_compile_complete"
+
+    - name: Compile ToDD
+      shell: ". /home/vagrant/.profile && make"
+      args:
+        chdir: "{{ todd_dir }}"
+      register: "todd_compiled"
+      when: not todd_compile_complete.stat.exists
+
+    - name: Marking Status of Compile ToDD as Complete
+      file:
+        path: "{{ todd_compile_check_file }}"
```

Comment:
Yeah I threw that /var/log in there as a placeholder as my original path would allow that check file to remain after a vagrant destroy which would keep those tasks from completing on another vagrant up. I originally used {{ todd_dir }} but /home/vagrant should work just fine.
I actually like to use a cleanup script when tearing down Vagrant environments which includes removing any remnants and etc. Ex. below...
````
#!/bin/bash
vagrant destroy -f
for file in *.retry; do
  if [[ -f $file ]]; then
    rm $file
  fi
done
if [ -d host_vars ]; then
  rm -rf host_vars
fi
if [ -d .vagrant ]; then
  rm -rf .vagrant
fi
if [ -f playbook.yml ]; then
  rm playbook.yml
fi
if [ -f Vagrantfile ]; then
  rm Vagrantfile
fi
````